AppCenter/AppCenter/Internals/Ingestion/MSHttpIngestion.m

```diff
@@ -222,11 +222,13 @@ - (void)sendCallAsync:(MSIngestionCall *)call {
       return;
     }
 
+    // Create a task for the request.
+    __block NSLock *taskLock = [[NSLock alloc] init];
```

Comment:
This code looks incorrect to me. I think what you do is basically create lock each time (each time code is here), so for each time it is referred on line #231, reference goes to separate monitor, which is meaningless. Correct me if im wrong.

Reply:
The phenomenon I observed was that every time a test project was run, a crash might occur.and the crash happens in this function, but run a test alone will not crash, so I think this is a problem with SDK concurrency, data confusion and data security issues when multiple threads access the same resource.Correct me if im wrong.